backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/bottomsheet/internal/BpkModalBottomSheetImpl.kt

```diff
@@ -0,0 +1,198 @@
+ * limitations under the License.
+ */
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+internal fun BpkModalBottomSheetImpl(
+    onDismissRequest: () -> Unit,
+    state: BpkModalBottomSheetState,
+    dragHandleStyle: BpkDragHandleStyle,
+    title: String?,
+    isClosable: BpkModalBottomSheetCloseAction,
+    content: @Composable ColumnScope.() -> Unit,
+    modifier: Modifier = Modifier,
+) {
+    ModalBottomSheet(
+        sheetState = state.delegate,
+        content = {
+            BottomSheetContent(dragHandleStyle = dragHandleStyle, content = {
+                BpkModalBottomSheetContent(
+                    dragHandleStyle = dragHandleStyle,
+                    title = title,
+                    isClosable = isClosable,
+                    state = state,
+                    onDismissRequest = onDismissRequest,
+                    content = content,
+                )
+            })
+        },
+        modifier = modifier,
+        dragHandle = { if (dragHandleStyle == BpkDragHandleStyle.Default) BpkBottomSheetHandle() },
+        shape = RoundedCornerShape(topStart = BpkBorderRadius.Lg, topEnd = BpkBorderRadius.Lg),
+        containerColor = BpkTheme.colors.surfaceElevated,
+        contentColor = BpkTheme.colors.textPrimary,
+        scrimColor = BpkTheme.colors.scrim,
+        tonalElevation = BpkElevation.Lg,
+        onDismissRequest = onDismissRequest,
+    )
+}
+
+@Composable
+fun ColumnScope.BpkModalBottomSheetContent(
+    dragHandleStyle: BpkDragHandleStyle,
+    title: String?,
+    isClosable: BpkModalBottomSheetCloseAction,
+    state: BpkModalBottomSheetState,
+    onDismissRequest: () -> Unit,
+    content: @Composable ColumnScope.() -> Unit,
+    modifier: Modifier = Modifier,
+) {
+    if (isClosable is BpkModalBottomSheetCloseAction.Default || !title.isNullOrEmpty()) {
+        when (dragHandleStyle) {
+            BpkDragHandleStyle.Default -> {
+                Column(modifier = modifier,
+                    verticalArrangement = Arrangement.spacedBy(BpkSpacing.Lg),
+                ) {
+                    BpkModalBottomSheetHeader(
+                        modifier = Modifier.height(BpkSpacing.Lg),
+                        title = title,
+                        state = state,
+                        dragHandleStyle = dragHandleStyle,
+                        isClosable = isClosable,
+                        onDismissRequest = onDismissRequest,
+                    )
+                    content()
+                }
+            }
+
+            is BpkDragHandleStyle.OnImage -> {
+                Box(modifier = modifier) {
+                    Column { content() }
+                    BpkModalBottomSheetHeader(
+                        title = title,
+                        state = state,
+                        dragHandleStyle = dragHandleStyle,
+                        isClosable = isClosable,
+                        onDismissRequest = onDismissRequest,
+                    )
```

Comment:
I had to wrap this with box so we make sure the header will be on top of the content

Reply:
I just noticed we have similar thing `BottomSheetContent`i will use that instead 😅